samza-core/src/main/java/org/apache/samza/storage/blobstore/BlobStoreBackupManager.java

```diff
@@ -139,8 +139,10 @@ public void init(Checkpoint checkpoint) {
 
     // Note: blocks the caller thread.
     // TODO LOW shesharma exclude stores that are no longer configured during init
+    // NOTE: Always get SnapshotIndex with getDeleted enabled. A failure to get a blob from SnapshotIndex would restart
```

Comment:
Minor: Comment isn't very clear. Doesn't explain _why_ this should be true. Only (if I understand correctly) what happens when this is true and there is still a failure.

Reply:
Sorry, the comment isn't accurate. Correcting this. We should pass false here and let the restart/restore recover so that we can get all blobs associated with this snapshot index restored as well.